### ISSUE TITLE
Problems with .equals

### ISSUE THREAD

**Issue by VioletFlare** [redis/redis-om-node#198]
Hi, I'm using redis-om 0.3.6 and I noticed a problem with equals.

Here's a screenshot of my DB:

![equals](https://github.com/redis/redis-om-node/assets/6873675/b760dc5a-dfde-49ed-8fdf-4f0ab90ba6d3)

Here's the code of the schema:

```const client = require('./Client');
const { Schema, Entity } = require('redis-om');

class NewsArticle extends Entity {};

const newsArticleSchema = new Schema(NewsArticle, {
    url: { type: 'string' }, 
    img: { type: 'string' }, 
    title: { type: 'text' },
    description: { type: 'text' },
    content: { type: 'text' },
    metaScrapedAtTimestamp: { type: 'date' },
    metaPublishedOnDiscord: { type: 'boolean' }
});

module.exports = client.then(c => {
    const newsArticleRepository = c.fetchRepository(newsArticleSchema);
    newsArticleRepository.createIndex();
    return newsArticleRepository;
});
```

I have this function which checks if there's already an article in my DB by checking the url.

```
checkIfArticleExists(url) {
        return newsArticleRepository.then(nar => {
            return nar.search().where('url').equals(url).return.count().then((count) => {
                if (!count) {
                    return false;
                } else {
                    return true;
                }
            });
        });
    }
```

While debugging it says it cannot find this record.

![equals2](https://github.com/redis/redis-om-node/assets/6873675/8a847f39-c65e-444d-8b79-e6f03b7e5f34)

Count is 0:

![image](https://github.com/redis/redis-om-node/assets/6873675/8187632c-48bd-4c0f-8711-3cd378f6d515)

Does the URL:

https://www.repubblica.it/esteri/2023/07/21/news/patrick_zaki_italia_gasparri_amnesty-408496548/?ref=RHLF-BG-I408498033-P2-S1-T1

Have something that Redis or Redis-OM might not like when it's defined as a string in the schema? Or is it me using the equals function wrong? I pretty much followed your tutorial on YT.

This function works for many different urls in my DB.

**Comment by guyroyse** [comment]
I suspect this is something in the URL that 0.3.6 doesn't escape or something like that. I'll see if I can reproduce and let ya know. Also, 0.4.0 is out and might be worth a look. The YouTube video doesn't use that version, but the README should cover everything you need to know.

**Comment by VioletFlare** [comment]
Hi, thanks for getting in touch this fast. I have removed the ?ref=RHLF-BG-I408498033-P2-S1-T1 portion from the URL and it seems to have worked... quite strange. Although I don't know how redis or the library works under the hood at all!

About the new version... I gave it a read before starting with the older version. I might upgrade to it soon :)

**Comment by guyroyse** [comment]
Try removing just the question mark. That’s my guess. I know that RediSearch changed a while back to add wildcards to tags. Might have introduced something I need to escape but didn’t in the past.

**Comment by VioletFlare** [comment]
Yeah, just removed the question mark, and now it works pretty well...
Coming from a totally different background (MySQL) I would never expect wildcards to be inside data.
Sounds like a bit of sorcery, haha.

**Comment by guyroyse** [comment]
I'll need to escape those for the next release. Let's keep this issue open until I do.

**Comment by guyroyse** [comment]
Resolved with PR #202. Published with Redis OM 0.4.2.
